app/controllers/pieces_controller.rb

```diff
@@ -16,30 +16,27 @@ def piece_params
   end
 
   def checks_before_move
-    return render plain: 'Invalid Move', status: :bad_request if flashmessages
+    flashmessages
     destination_x = piece_params[:column_coordinate].to_i
     destination_y = piece_params[:row_coordinate].to_i
     render plain: 'Invalid Move', status: :bad_request if @piece.not_moved?(destination_x, destination_y)
     if !@piece.obstructed?(destination_x, destination_y) && @piece.valid_move?(destination_x, destination_y) && check_turn == @piece.color
       change_turn
       @piece.move_to!(destination_x, destination_y)
-      check_game_status
+      current_game.set_game_status
+      flashmessages
       redirect_to game_path(@piece.game_id)
     else
       render plain: 'Invalid Move', status: :bad_request
     end
   end
 
-  def check_game_status
-    @piece.game.update(turn: -1) if @piece.game.checkmate?
-    @piece.game.update(turn: -2) if @piece.game.stalemate?
-    flashmessages
-  end
-
   def flashmessages
-    return flash[:notice] = 'Checkmate !!' if @piece.game.turn == -1
-    return flash[:notice] = 'Stalemate !!' if @piece.game.turn == -2
-    false
+    return flash[:notice] = 'Checkmate !!' if current_game.game_status == 1
+    return flash[:notice] = 'Stalemate !!' if current_game.game_status == 2
+    return flash[:notice] = 'Check !!' if current_game.game_status == 3
+    return flash[:notice] = 'Draw !!' if current_game.game_status == 4
```

Comment:
Oh I kept reading. I see you had as an enum. Was it removed because the names overlapped with the method names?